sympy/core/expr.py

```diff
@@ -2695,7 +2695,7 @@ def is_rational_function(self, *syms):
         See also is_algebraic_expr().
 
         """
-        if self in _illegal:
+        if self.has_xfree(set(_illegal)):  # zoo/x
             return False
```

Comment:
What does this do? The `is_rational_function` method already walks through the expression tree. Won't this cause it to walk the tree again and again for each node in the tree?